lib/Controller/SocialApiController.php

```diff
@@ -0,0 +1,294 @@
+
+			// search for this network in user's profile
+			foreach ($socialentry as $networkentry => $profileId) {
+				if ($network === strtolower($networkentry)) {
+					// cleanups
+					if (in_array('basename', $social['cleanups'])) {
+						$profileId = basename($profileId);
+					}
+					if (in_array('filter', $social['cleanups'])) {
+						if (preg_match($social['filter']['regex'], $profileId, $matches)) {
+							$profileId = $matches[$social['filter']['group']];
+						}
+					}
+					// checks
+					if (in_array('number', $social['checks'])) {
+						if (!ctype_digit($profileId)) {
+							break;
+						}
+					}
+					$connector = str_replace("{socialId}", $profileId, $social['recipe']);
+					break;
+				}
+			}
+			if ($connector) {
+				break;
+			}
+		}
+		return ($connector);
+	}
+
+
+	/**
+	 * @NoAdminRequired
+	 *
+	 * Retrieves social profile data for a contact
+	 *
+	 * @param {String} addressbookId the addressbook identifier
+	 * @param {String} contactId the contact identifier
+	 * @param {String} type the kind of information to retrieve -- provision
+	 *
+	 * @returns {JSONResponse} an empty JSONResponse with respective http status code
+	 */
+	public function fetch(string $addressbookId, string $contactId, string $type) : JSONResponse {
+
+		$url = null;
+
+		try {
+			// get corresponding addressbook
+			$addressBooks = $this->manager->getUserAddressBooks();
+			$addressBook = null;
+			foreach($addressBooks as $ab) {
+				if ($ab->getUri() === $addressbookId) {
+					$addressBook = $ab;
+				}
+			}
+			if (is_null($addressBook)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+			}
+
+			// search contact in that addressbook
+			$contact = $addressBook->search($contactId, ['UID'], [])[0];
+			if (is_null($contact)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR); 
+			}
+
+			// get social data
+			$socialprofile = $contact['X-SOCIALPROFILE'];
+
+			if (is_null($socialprofile)) {
+				return new JSONResponse([], Http::STATUS_INTERNAL_SERVER_ERROR);
+			}
+
+			// retrieve data
+			try {
+				$url = $this->getSocialConnector($socialprofile);
+			}
+			catch (Exception $e) {
+				return new JSONResponse([], Http::STATUS_BAD_REQUEST);
+			}
+
+			if (empty($url)) {
+				return new JSONResponse([], Http::STATUS_NOT_IMPLEMENTED);
+			}
+
+			$host = parse_url($url);
+			if (!$host) {
+				return new JSONResponse([], Http::STATUS_BAD_REQUEST);
+			}
+			$opts = [
+				"http" => [
+					"method" => "GET",
+					"header" => "User-Agent: Nextcloud Contacts App"
+				]
+			];
+			$context = stream_context_create($opts);
+			$socialdata = file_get_contents($url, false, $context);
+
+			$image_type = null;
+			foreach ($http_response_header as $value) {
+				if (preg_match('/^Content-Type:/i', $value)) {
+					if (stripos($value, "image") !== false) {
+						$image_type = substr($value, stripos($value, "image"));
+					}
+				}
+			}
+
+			if ((!$socialdata) || ($image_type === null)) {
```

Comment:
```suggestion
			if (!$socialdata || $image_type === null) {
```
Unnecessary parenthesis?